cmd/update/volume_move.go

```diff
@@ -0,0 +1,383 @@
+			params.CASVersion = version.FromString(c.casVersion)
+		}
+
+		err = c.client.AddDeploymentOnNode(ctx, nsID, vol.ID, params)
+		if err != nil {
+			return fmt.Errorf("failed to add deployment on node %s: %s", c.toNode, err)
+		}
+	}
+
+	// wait for a new replica on [destination] node
+	err = c.waitForReplicaOnNode(ctx, nsID)
+	if err != nil {
+		return err
+	}
+
+	// get updated volume config
+	vol, err = c.getVolume(ctx, nsID)
+	if err != nil {
+		return fmt.Errorf("failed to get updated volume config from api: %s", err)
+	}
+
+	if !isMovingMaster {
+		// mark the replica on the [source] node as preferred for eviction
+		//
+		// the master node needs to first be moved and then be evicted due to
+		// api endpoints restriction
+		params := &apiclient.SetPreferredEvictionCandidatesRequestParams{
+			DeploymentIDs: []string{deploymentOnSourceNode.String()},
+		}
+		if c.useCAS() {
+			params.CASVersion = vol.Version
+		}
+		err = c.client.SetPreferredEvictionCandidates(ctx, nsID.String(), vol.ID.String(), params)
+		if err != nil {
+			return fmt.Errorf("failed to set preferred eviction candidates: %s", err)
+		}
+
+		c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica %s on [source] node %s marked as preferred for eviction", deploymentOnSourceNode, c.fromNode))
+	}
+
+	// wait for the new replica on [destination] node to sync
+	err = c.waitForReplicaOnNodeToSync(ctx, nsID)
+	if err != nil {
+		return err
+	}
+
+	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [destination] node %s finished syncing", c.toNode))
+
+	if !isMovingMaster {
+		_ = c.client.EvictReplica(ctx, nsID.String(), vol.ID.String(), deploymentOnSourceNode.String())
+		if err != nil {
+			if !errors.Is(err, apiclient.VolumeNotFoundError{}) {
+				return fmt.Errorf("failed to force evict replica from [source] node %s (will eventually converge): %w", c.fromNode, err)
+			}
+		}
+
+		c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [source] node %s successfully evicted", c.fromNode))
+		return nil
+	}
+
+	// from here on, we are attempting to move a master
+
+	// find the deployment IDs of the newly created replica and the master being moved
+	var newReplicaID id.Deployment
+	for _, rep := range vol.Replicas {
+		if rep.Node.String() == c.toNode {
+			newReplicaID = rep.ID
+		}
+	}
+	oldMasterID := vol.Master.ID
+
+	err = c.client.AttemptPromotion(ctx, string(nsID), vol.ID.String(), newReplicaID.String())
+	if err != nil {
+		return err
+	}
+
+	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [destination] node %s attempting promotion to master", c.toNode))
+
+	err = c.waitForMasterPromote(ctx, nsID)
+	if err != nil {
+		return err
+	}
+
+	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [destination] node %s successfully promoted to master", c.toNode))
+
+	err = c.client.EvictReplica(ctx, nsID.String(), vol.ID.String(), oldMasterID.String())
+	if err != nil {
+		if !errors.Is(err, apiclient.VolumeNotFoundError{}) {
```

Comment:
What happens if the error is `apiclient.VolumeNotFoundError`, isn't that also a failure case? If the volume's been deleted which should log

Reply:
Since the point of this step is to get rid of one deployment, failing to find it is seen as an ok response here (whatever reason for that may be)

Reply:
Ah is `VolumeNotFoundError` the error returned when a deployment is not found?